rllib/core/learner/learner.py

```diff
@@ -1228,13 +1228,16 @@ def additional_update_for_module(
 
     def update(
         self,
-        batch: MultiAgentBatch,
         *,
-        minibatch_size: Optional[int] = None,
-        num_iters: int = 1,
+        batch: Optional[MultiAgentBatch] = None,
```

Comment:
Happy to discuss the alternative to provide two different (mutually exclusive?) methods that the user/algo can decide to call: `update_from_batch` (for algos that do NOT require episode processing, such as DQN) or `update_from_episodes` (for algos that require a view on the sampled episodes for e.g. vf-bootstrapping, vtrace, etc..).

Reply:
I like it if the two methods are separated. I don't think there would be a case where a specific algorithm's learner would have both methods implemented. ie DQN will only implemented `update_from_batch`, and PPO would only implement `update_from_episodes`. This is much much cleaner than mixing both into one function. The user will have to deal with less cognitive load if they are separated.

Reply:
I separated them in the LearnerGroup and Learner APIs:
- update_from_batch(async=False|True)
- update_from_episodes(async=False|True)

Reply:
Also, I think it's nicer to have the `async_update` bool option as an extra arg (instead of separate method) for better consistency and less code bloat.